test/e2e/e2e.go

```diff
@@ -40,3 +45,73 @@ func SetupWithNamespace(t *testing.T, namespace string) *test.Clients {
 	}
 	return clients
 }
+
+// CreateNamespace creates a namespace to run the integration tests.
+func CreateNamespace(t *testing.T, clients *test.Clients, namespace string) {
+	_, err := clients.KubeClient.Kube.CoreV1().Namespaces().Get(namespace, metav1.GetOptions{})
+	if err != nil && apierrs.IsNotFound(err) {
+		// Create the namespace if not available
+		if _, err = clients.KubeClient.Kube.CoreV1().Namespaces().Create(&corev1.Namespace{ObjectMeta:
+			metav1.ObjectMeta{Name: namespace}}); err != nil {
+			t.Fatalf("Failed to create the namespace %s: %v", namespace, err)
+		}
+	}
+}
+
+// TestKnativeServingCreation creates a KnativeServing to see if it can reach the READY status.
+func TestKnativeServingCreation(t *testing.T, clients *test.Clients, names test.ResourceNames) {
```

Comment:
Why are these functions big top-level package test helpers vs. just helpers in the test file itself?

Reply:
I renamed them